test/solidity/modifiers/Modifiers.js

```diff
@@ -0,0 +1,139 @@
+        });
+    
+        const finalBalance = await modifiersContract.getBalance();
+        expect(tinybarToWeibar(finalBalance.add(initialBalance))).to.equal(paymentAmount);
+    });    
+
+    it("Should have the correct MAX_SUPPLY value", async function() {
+        const maxSupply = await modifiersContract.MAX_SUPPLY();
+        expect(maxSupply).to.equal(1000000);
+    });
+
+    it("Should set deploymentTimestamp to the block timestamp of deployment", async function() {
+        const txReceipt = await modifiersContract.deployTransaction.wait();
+        const block = await ethers.provider.getBlock(txReceipt.blockHash);
+
+        const deploymentTimestamp = await modifiersContract.deploymentTimestamp();
+        expect(deploymentTimestamp).to.equal(block.timestamp);
+    });
+
+    it("Should emit indexed from and to values in the RegularEvent", async function() {
+        const toAddress = accounts[1].address;
+        const tx = await modifiersContract.triggerRegularEvent(toAddress, 100, "test transfer");
+        const receipt = await tx.wait();
+    
+        expect(receipt.events?.length).to.equal(1);
+        const event = receipt.events[0];
+    
+        // Check the event's topics. The first topic is the event's signature.
+        // The next topics are the indexed parameters in the order they appear in the event.
+        expect(event.topics[1].toLowerCase()).to.equal(ethers.utils.hexZeroPad(accounts[0].address, 32).toLowerCase()); // from address
+        expect(event.topics[2].toLowerCase()).to.equal(ethers.utils.hexZeroPad(toAddress, 32).toLowerCase()); // to address
+    });
+
+    it("Should emit the AnonymousEvent with correct values", async function() {
+        const tx = await modifiersContract.triggerAnonymousEvent(257);
+        const receipt = await tx.wait();
+    
+        expect(receipt.events?.length).to.equal(1);
+    
+        const anonymousEvent = receipt.events[0];
+        expect(anonymousEvent.event).to.undefined;
+
+        // Since it's anonymous, we access the topics directly to get the indexed values.
+        const senderAddress = "0x" + anonymousEvent.topics[0].slice(-40);
+        expect(senderAddress.toLowerCase()).to.equal(accounts[0].address.toLowerCase());
+        const value = ethers.utils.defaultAbiCoder.decode(["uint256"], anonymousEvent.data);
+        expect(value[0]).to.equal(257);
+    });    
+
+    it("Should return the message in the from the derived contract that overrides the virtual function", async function() {
+        expect(await derivedContract.getData()).to.equal(55);
+        expect(await derivedContract.show()).to.equal("This is the derived contract");
+    });
+    
+    it("Should return the message in the from ContractB that overrides the show function", async function() {
+        expect(await contractB.getData()).to.equal(79);
+        expect(await contractB.show()).to.equal("This is the overriding contract B");
+    });    
+    
```

Comment:
nit: a few of these have an extra line before the final closing bracket

Reply:
Removed.